kube-runtime/src/utils/event_flatten.rs

```diff
@@ -9,22 +9,17 @@ use pin_project::pin_project;
 #[pin_project]
 /// Stream returned by the [`applied_objects`](super::WatchStreamExt::applied_objects) and [`touched_objects`](super::WatchStreamExt::touched_objects) method.
 #[must_use = "streams do nothing unless polled"]
-pub struct EventFlatten<St, K> {
+pub struct EventFlatten<St> {
     #[pin]
     stream: St,
     emit_deleted: bool,
-    queue: std::vec::IntoIter<K>,
 }
-impl<St: TryStream<Ok = Event<K>>, K> EventFlatten<St, K> {
+impl<St: TryStream<Ok = Event<K>>, K> EventFlatten<St> {
     pub(super) fn new(stream: St, emit_deleted: bool) -> Self {
-        Self {
-            stream,
-            queue: vec![].into_iter(),
-            emit_deleted,
-        }
```

Comment:
kind of cool that this got simplified as a result

Reply:
yeah, think we can maybe deprecate / repurpose this thing entirely to some kind of filter module instead (since the concept of an "unflattened stream" kind of goes away externally (which is good, it was a big source of confusion).